view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java

```diff
@@ -488,12 +506,32 @@ private static List<TierConfig> getTableTierConfigs(@Nullable PinotTableSpec tab
   }
 
   private static List<FieldConfig> toFieldConfigs(@Nullable PinotTableSpec tableSpec) {
-    if (tableSpec == null || tableSpec.getTextIndexColumns() == null) {
+    if (tableSpec == null || tableSpec.getTextIndexConfigs() == null) {
       return null;
     }
 
-    return tableSpec.getTextIndexColumns().stream()
-        .map(columnName -> new FieldConfig(columnName, EncodingType.RAW, IndexType.TEXT, null))
+    return tableSpec.getTextIndexConfigs().stream()
+        .map(
+            textIndexConfig -> {
+              String columnName = textIndexConfig.getString(TEXT_INDEX_CONFIG_COLUMN);
+              FieldConfig fieldConfig =
```

Comment:
Instead of first creating FieldConfig and then overriding it, we can create it conditionally in separate if-else.
```
if (textIndexConfig.hasPath(TEXT_INDEX_CONFIG_SKIP_PRIOR_SEGMENTS)) {
 FieldConfig fieldConfig = new FieldConfig(with map);
} else {
 FieldConfig fieldConfig = new FieldConfig(without map); 
}
```

Reply:
Addressed